src/managers/GuildInviteManager.js

```diff
@@ -0,0 +1,187 @@
+   * @example
+   * // Fetch a single invite
+   * guild.invites.fetch('bRCvFy9')
+   *   .then(console.log)
+   *   .catch(console.error);
+   * @example
+   * // Fetch a single invite without checking cache
+   * guild.invites.fetch({ code: 'bRCvFy9', force: true })
+   *   .then(console.log)
+   *   .catch(console.error)
+   * @example
+   * // Fetch a single invite without caching
+   * guild.invites.fetch({ code: 'bRCvFy9', cache: false })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  fetch(options) {
+    if (!options) return this._fetchMany();
+    if (typeof options === 'string') {
+      const code = DataResolver.resolveInviteCode(options);
+      if (!code) return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+      return this._fetchSingle({ code, cache: true });
+    }
+    if (!options.code) {
+      if ('cache' in options) return this._fetchMany(options.cache);
+      return Promise.reject(new Error('INVITE_RESOLVE_CODE'));
+    }
+    return this._fetchSingle({
+      ...options,
+      code: DataResolver.resolveInviteCode(options.code),
+    });
+  }
+
+  async _fetchSingle({ code, cache, force = false }) {
+    if (!force) {
+      const existing = this.cache.get(code);
+      if (existing) return existing;
+    }
+
+    const invites = await this._fetchMany(cache);
+    const invite = invites.get(code);
+    if (!invite) throw new Error('INVITE_NOT_FOUND');
+    return invite;
+  }
+
+  async _fetchMany(cache) {
+    const data = await this.client.api.guilds(this.guild.id).invites.get();
+    return data.reduce((col, invite) => col.set(invite.code, this.add(invite, cache)), new Collection());
+  }
+
+  /**
+   * @typedef {CreateInviteOptions} CreateChannelInviteOptions
+   * @property {GuildChannelResolvable} [channel] The optional channel to define where to create the invite.
+   */
+
+  /**
+   * Create a invite to the guild from the provided or the first available channel.
+   * @param {CreateChannelInviteOptions} [options={}] The options for creating the invite from a optional channel.
+   * @returns {Promise<Invite>}
+   * @example
+   * // Create an invite to a random allowed channel
+   * guild.invites.create()
+   *   .then(invite => console.log(`Created an invite with a code of ${invite.code} from ${invite.channel}`))
+   *   .catch(console.error);
+   * @example
+   * // Create an invite to a selected channel
+   * guild.invites.create({ channel: '599942732013764608' })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  create(options = {}) {
+    let channel;
+
+    if (options.channel) {
+      channel = this.guild.channels.resolve(options.channel);
+      if (!channel) return Promise.reject(new Error('GUILD_CHANNEL_RESOLVE'));
+    } else {
+      channel = this.guild.channels.cache.find(c =>
+        c.permissionsFor(this.guild.me).has(Permissions.FLAGS.CREATE_INSTANT_INVITE),
+      );
+      if (!channel) return Promise.reject(new Error('GUILD_CHANNEL_INVITE'));
```

Comment:
Picking a more or less arbitrary channel to create an invite to is a bit out-of-place with how the library operates.

Arbitrary as in that the order of the channels in this collection is undefined (technically insertion order by how Discord happens to send them to use and channels get created while the bot is running and not establishing a new session).

---

The developer might want to create one on a text channel, but this could also be creating one on a voice channel. (And vice-versa)
They might be expecting the top-most channel to be picked, they might want to pick a "general" or "welcome" channel, if one exist. There probably are a few more common things about this, which I can't think of right now.

---

This behaviour also makes you lose on error-handling, you couldn't distinguish between those, without checking yourself too:
- Was there no channel at all?
- No channel of the correct type?
- No channel with permissions?
- Do we not have permissions to create invites guild-wide?

---

We shouldn't make those assumptions and let the developer handle that themselves by making the `channel` parameter mandatory.

Reply:
> Is there any benefit (except backward-compatibility) in keeping `Guild#fetchInvites`?
> If not, that method should be removed with this PR.

Like `Guild#fetchBans` when adding the `GuildBanManager` I agree that it should be removed 👍

Reply:
> Arbitrary as in that the order of the channels in this collection is undefined (technically insertion order by how Discord happens to send them to use and channels get created while the bot is running and not establishing a new session).

We could sort them from the `GuildChannel#rawPosition` 

> The developer might want to create one on a text channel, but this could also be creating one on a voice channel. (And vice-versa)

The primary interest of this functionality is to reach a targeted server.
  - When we want to join a voice channel, we target the voice channel in the functionality that links the user to the room.
  - When you simply want to join a targeted server, the goal is to get to the server, regardless of the channel, but not by joining a voice room.

This is also why Discord itself has integrated an interface that takes the first textual room available to invite a user from your direct messages (without even entering an error when this is not possible).

![image](https://user-images.githubusercontent.com/29425008/123560263-78b89000-d7a1-11eb-9805-7486355e9516.png)
![image](https://user-images.githubusercontent.com/29425008/123560470-989c8380-d7a2-11eb-8e23-7c91e3e66c5c.png)

> This behavior also makes you lose on error-handling, you couldn't distinguish between those, without checking yourself too:
> 
> - Was there no channel at all?
> - No channel of the correct type?
> - No channel with permissions?
> - Do we not have permissions to create invites guild-wide?

I think we should work on these points to make this possible properly.

> Picking a more or less arbitrary channel to create an invite to is a bit out-of-place with how the library operates.

I think this feature aims not to pick a random channel but rather to get an invite to the targeted server. And at present and with the means offered by the Discord API, this is not possible otherwise. As shown above, this is how Discord does it. 

I think that this library should cover an action as simple as getting the invitation from a server that the bot is part of instead of leaving it to its developer given the difficulty that this represents and all the "hidden" and "complex" aspects that this includes.

I wouldn't object to removing it if you think it has no place in the library but I still wanted to explain my opinion.

Reply:
We already figured that was the case, but we still believe firmly in our stance here.

Reply:
Okay 👍

Reply:
Maybe in the future we can revisit this, but as it currently stands we'd rather move forward without it, instead of blocking the whole PR because of it 👍

Reply:
Should we roll back to independent channel params?

Reply:
cc @SpaceEEC

Reply:
I think that's similar to how we do this in `MessageManager` and `GuildMemberManager`, so I'd be in favor of separate parameters.